packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts

```diff
@@ -291,7 +291,9 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Ethereum> {
 
   async validateAddress(address: string): Promise<chainAdapters.ValidAddressResult> {
     const isValidAddress = WAValidator.validate(address, this.getType())
-    if (isValidAddress) return { valid: true, result: chainAdapters.ValidAddressResultType.Valid }
+    const isValidEnsAddress = new RegExp(/([A-Za-z]*\.)?([A-Za-z]{3,})\.eth/).test(address)
```

Comment:
- `/foobar/` is a regex literal, so no need to call `new RegExp()`
- need to match the whole string
- regex now respects RFC 1123
```suggestion
    const isValidEnsAddress = /^([0-9A-Z]([-0-9A-Z]*[0-9A-Z])?\.)+eth$/i.test(address)
```

Reply:
Nice, thanks! Before I commit these improvements, regarding RFC 1123, what do you think bringing back the `{3,}` repeater on the last group?
ENS has at least one gotcha compared to it
If you try to register a sub/domain name on ENS where the domain is less than 3 characters long, you will have an error:
https://app.ens.domains/name/test.vi.eth 🚫 
https://app.ens.domains/name/test.vit.eth/subdomains ✅

Reply:
Did some looking at the ENS contracts to hunt this down -- I'm pretty sure that's an artificial restriction. There's nothing preventing names from being shorter except that the current registrar controller for the `.eth` domain won't register them for you. But if you had one, it would totally be resolvable, and even transferable and renewable. So IMO we should leave the regex open-structured since future controller implementations relax that restriction.

Reply:
That's probably why I didn't find anything on the spec about it then. 
If it's something registar-specific and it's valid according to , I agree it should be returned as a valid address, because it is.

Reply:
do we want to consider supporting all TLDs as this is now supported by ENS, or is that too premature?

https://medium.com/the-ethereum-name-service/how-eth-names-compare-to-dns-names-imported-to-ens-ab855a6dbdb2

Reply:
Ohhh that's a valid point. TBH I haven't seen a single DNS domain in the wild ever (that is, someone posting their regular DNS domain imported in ENS with a message like "guys this is my address).
But if it's valid it's valid, I'd say let's add it

Reply:
I say it's premature. I'd want a curated whitelist of TLDs we're willing to resolve, given that they have different security properties than .eth, and the regex is a handy point for expanding those in the future.